components/search/src/types.rs

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// The list of possible application names that are currently supported.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchApplicationName {
+    Firefox = 1,
+    FirefoxAndroid = 2,
+    FocusAndroid = 3,
+    FirefoxIOS = 4,
+    FocusIOS = 5,
+}
+
+impl SearchApplicationName {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SearchApplicationName::Firefox => "firefox",
+            SearchApplicationName::FirefoxAndroid => "firefox-android",
+            SearchApplicationName::FocusAndroid => "focus-android",
+            SearchApplicationName::FirefoxIOS => "firefox-ios",
+            SearchApplicationName::FocusIOS => "focus-ios",
+        }
+    }
+}
+
+/// The list of possible update channels for a user's build.
+/// Use `default` for a self-build or an unknown channel.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchUpdateChannel {
+    Default,
+    Nightly,
+    Aurora,
+    Beta,
+    Release,
+    ESR,
+}
+
+impl SearchUpdateChannel {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SearchUpdateChannel::Default => "default",
+            SearchUpdateChannel::Nightly => "nightly",
+            SearchUpdateChannel::Aurora => "aurora",
+            SearchUpdateChannel::Beta => "beta",
+            SearchUpdateChannel::Release => "release",
+            SearchUpdateChannel::ESR => "esr",
+        }
+    }
+}
+
+/// The user's environment that is used for filtering the search configuration.
+#[derive(Debug, uniffi::Record)]
+pub struct SearchUserEnvironment {
+    /// The current locale of the application that the user is using.
+    pub locale: String,
+
+    /// The home region that the user is currently identified as being within.
+    /// On desktop & android there is a 14 day lag after detecting a region
+    /// change before the home region changes. TBD: iOS?
+    pub region: String,
+
+    /// The update channel of the user's build.
+    pub update_channel: SearchUpdateChannel,
+
+    /// The distribution id for the user's build.
+    pub distribution_id: String,
+
+    /// The search related experiment id that the user is included within. On
+    /// desktop this is the `searchConfiguration.experiment` variable.
+    pub experiment: String,
```

Comment:
I am not sure what this is for Android today. Will look for it

Reply:
I don't think Android has anything similar to this. We introduced it on desktop with the previous version of the configuration because it was that configuration was flexible enough to allow the experimentation with the different parameters & deployments.